File: ios/EventEmitter/EventEmitter.m
//
//  Events.m
//  Obsidian
//
//  Created by Tommaso Carpi on 30/01/21.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(EventEmitter, RCTEventEmitter)
  RCT_EXTERN_METHOD(supportedEvents)
@end

File: ios/EventEmitter/EventEmitter.swift
//
//  EventEmitter.swift
//  Obsidian
//
//  Created by Tommaso Carpi on 30/01/21.
//

import Foundation

@objc(EventEmitter)
open class EventEmitter: RCTEventEmitter {

  public static var emitter: RCTEventEmitter!
  
  @objc public override static func requiresMainQueueSetup() -> Bool {return true}

  override init() {
    super.init()
    EventEmitter.emitter = self
  }

  open override func supportedEvents() -> [String] {
    ["url"]
  }
  
}


